core/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -300,6 +300,15 @@ public Builder addBlocks(IndexMetaData indexMetaData) {
             return this;
         }
 
+        public Builder updateBlocks(IndexMetaData indexMetaData) {
+            removeIndexBlock(indexMetaData.index(), MetaDataIndexStateService.INDEX_CLOSED_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_READ_ONLY_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_READ_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_WRITE_BLOCK);
+            removeIndexBlock(indexMetaData.index(), IndexMetaData.INDEX_METADATA_BLOCK);
+            return addBlocks(indexMetaData);
+        }
+
```

Comment:
The name of the method confuses me a bit because it's not an "update" method but rather a "remove specific blocks"  + "add" method. 

This seems to be specific to some index blocks in a given context (restoring an index) and this method is located in `ClusterBlocks.Builder` class... I think the removals should be explicitly located in the `RestoreService` instead.